crates/wasmtime/src/runtime/module.rs

```diff
@@ -1027,11 +1027,42 @@ impl Module {
         })
     }
 
+    /// Get the locations of functions in this module's `.text` section, with the
+    /// function names.
+    ///
+    /// Each function's location is a (name, `.text` section offset, length) pair.
+    pub fn function_locations_with_names<'a>(
+        &'a self,
+    ) -> impl ExactSizeIterator<Item = FunctionLocationWithName> + 'a {
```

Comment:
I think `function_loactions` is only used by this method, so we can we remove it or make it non-`pub`, as I suggested previously?

Also I like the named struct with named fields, that's indeed better than a large-ish tuple, but the docs here should match that now.

And since this is really more than just function locations and names (it also has an index, for example) maybe we should call this method `function_metadatas` and the struct `FunctionMetadata`? Feel free to take-or-leave this last one...

Reply:
Good catch, let me clean this up.